benchmarks/helics/pholdBenchmarks.cpp

```diff
@@ -0,0 +1,197 @@
+        std::vector<PholdFederate> feds (fed_count);
+        for (int ii = 0; ii < fed_count; ++ii)
+        {
+            feds[ii].initialize (wcore->getIdentifier (), ii, fed_count);
+        }
+
+        std::vector<std::thread> threadlist (static_cast<size_t> (fed_count-1));
+        for (int ii = 0; ii < fed_count-1; ++ii)
+        {
+            threadlist[ii] = std::thread ([&] (PholdFederate &f) { f.run ([&brr] () { brr.wait (); }); },
+                                          std::ref (feds[ii+1]));
+        }
+        feds[0].makeReady ();
+        brr.wait ();
+        state.ResumeTiming ();
+        feds[0].run ([] () {});
+        state.PauseTiming ();
+        for (auto &thrd : threadlist)
+        {
+            thrd.join ();
+        }
+
+        int totalEvCount = 0;
+        for (int ii = 0; ii < fed_count; ++ii)
+        {
+            totalEvCount += feds[ii].evCount;
+        }
+        state.counters["EvCount"] = totalEvCount;
+ 
+        wcore.reset ();
+        cleanupHelicsLibrary ();
+        state.ResumeTiming ();
+    }
+}
+// Register the function as a benchmark
+BENCHMARK (BM_phold_singleCore)
+  ->RangeMultiplier (2)
+  ->Range (1, 16)
+  ->Unit (benchmark::TimeUnit::kMillisecond)
+  ->Iterations (1)
+  ->UseRealTime ();
+
+static void BM_phold_multiCore (benchmark::State &state, core_type cType)
+{
+    for (auto _ : state)
+    {
+        state.PauseTiming ();
+
+        int fed_count = static_cast<int> (state.range (0));
+        gmlc::concurrency::Barrier brr (static_cast<size_t>(fed_count));
+
+        auto broker = helics::BrokerFactory::create (cType, "brokerb",
+                                                     std::string ("--federates=") + std::to_string (fed_count));
+        broker->setLoggingLevel (helics_log_level_no_print);
+        auto wcore = helics::CoreFactory::create (cType, std::string ("--federates=1"));
+        std::vector<PholdFederate> feds (fed_count);
+        std::vector<std::shared_ptr<helics::Core>> cores (fed_count);
+        for (int ii = 0; ii < fed_count; ++ii)
+        {
+            cores[ii] = helics::CoreFactory::create (cType, "-f 1");
+            cores[ii]->connect ();
+            feds[ii].initialize (cores[ii]->getIdentifier (), ii, fed_count);
+        }
+
+        std::vector<std::thread> threadlist (static_cast<size_t> (fed_count-1));
+        for (int ii = 0; ii < fed_count-1; ++ii)
+        {
+            threadlist[ii] = std::thread ([&] (PholdFederate &f) { f.run ([&brr] () { brr.wait (); }); },
+                                          std::ref (feds[ii+1]));
+        }
+        feds[0].makeReady ();
+        brr.wait ();
+        state.ResumeTiming ();
+        feds[0].run ([] () {});
+        state.PauseTiming ();
+        for (auto &thrd : threadlist)
+        {
+            thrd.join ();
+        }
+
+        int totalEvCount = 0;
+        for (auto &f : feds)
+        {
+            totalEvCount += f.evCount;
```

Comment:
what is this counting?

Reply:
The total number of events, since typically the PHOLD benchmark is events/second handled.